### PR TITLE
tcpblaster reporting and documentation improvements

### PR DESCRIPTION
**Pull request by adamfeuer**
### Summary

* Improves reporting with timestamp, group count, and clarified bandwidth (changes message that incorrectly shows Kb [kilobits] to KB [kilobytes])
* Adds group configuration (was hardcoded at 50 buffers per send; this is now configurable)
* Adds README that shows how to configure and run

### Impact

* None if you don't use tcpblaster
* tcpblaster is now more configurable and has clearer reporting that allows for hang-detection on long runtimes

### Testing

* Manual on the simulator, SAMA5D3-Xplained (SAMA5D36), and Linux.

### How To Verify 

* Run tcpblaster as per the README

### CODE REVIEW

File: examples/tcpblaster/tcpblaster_client.c [hunk 1]
```diff
@@ -218,10 +224,13 @@ void tcpblaster_client(void)
               elapsed.tv_nsec = curr.tv_nsec + borrow;
             }
 
+          char timebuff[100];
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_client.c]: Need move to variable declaration section

File: examples/tcpblaster/tcpblaster_client.c [hunk 2]
```diff
@@ -218,10 +224,13 @@ void tcpblaster_client(void)
...
+          strftime (timebuff, 100, "%Y-%m-%d %H:%M:%S.000", localtime (&curr));
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_client.c]: revmove space before (

File: examples/tcpblaster/tcpblaster_server.c [hunk 3]
```diff
@@ -247,13 +251,17 @@ void tcpblaster_server(void)
               elapsed.tv_nsec = curr.tv_nsec + borrow;
             }
 
+          char timebuff[100];
+          strftime (timebuff, 100, "%Y-%m-%d %H:%M:%S.000", localtime (&curr));
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_server.c]: Same comment as before

File: examples/tcpblaster/tcpblaster_server.c [hunk 4]
```diff
@@ -247,13 +251,17 @@ void tcpblaster_server(void)
...
+
           fkbsent  = (float)recvtotal / 1024.0;
           felapsed = (float)elapsed.tv_sec + (float)elapsed.tv_nsec / 1000000000.0;
-          printf("Received %d buffers:  %7.1f Kb (avg %5.1f Kb) in %6.2f Sec (%7.1f Kb/Sec)\n",
-                  recvcount, fkbsent, fkbsent/recvcount, felapsed, fkbsent/felapsed);
+            printf("[%s] %d: Received %d buffers: %7.1f KB (buffer average size: %5.1f KB) in %6.2f seconds (%7.1f KB/second)\n",
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_server.c]: Remove two space before printf

File: examples/tcpblaster/tcpblaster_server.c [hunk 5]
```diff
@@ -257,7 +257,7 @@ void tcpblaster_server(void)
 
           fkbsent  = (float)recvtotal / 1024.0;
           felapsed = (float)elapsed.tv_sec + (float)elapsed.tv_nsec / 1000000000.0;
-            printf("[%s] %d: Received %d buffers: %7.1f KB (buffer average size: %5.1f KB) in %6.2f seconds (%7.1f KB/second)\n",
+          printf("[%s] %d: Received %d buffers: %7.1f KB (buffer average size: %5.1f KB) in %6.2f seconds (%7.1f KB/second)\n",
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_server.c]: Let's merge into one patch instead

File: examples/tcpblaster/tcpblaster_client.c [hunk 6]
```diff
@@ -166,7 +166,6 @@ void tcpblaster_client(void)
 
       /* Wait until we can send data or until the connection is lost */
 
-      printf("=");
```
Comment (xiaoxiang781216) [inline examples/tcpblaster/tcpblaster_client.c]: Modify the previous patch instead, it's better to make the patch clean if it doesn't merge into mainline yet.

File: examples/tcpblaster/tcpblaster_client.c (hunk 1 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_client.c]: @xiaoxiang781216 will fix

File: examples/tcpblaster/tcpblaster_client.c (hunk 2 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_client.c]: @xiaoxiang781216 will fix

File: examples/tcpblaster/tcpblaster_server.c (hunk 3 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_server.c]: @xiaoxiang781216 will fix

File: examples/tcpblaster/tcpblaster_server.c (hunk 4 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_server.c]: @xiaoxiang781216 will fix

File: examples/tcpblaster/tcpblaster_server.c (hunk 5 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_server.c]: @xiaoxiang781216 Do you mean, squash all the commits into one so master doesn't end up with multiple commits?

Comment (xiaoxiang781216) [reply examples/tcpblaster/tcpblaster_server.c]: No, it doesn't make sense to request all patches merge into one. But in one PR, if the first patch has some issue, and the second patch fix that issue(.e.g printf alignment, printf("=")), we should merge the second patch into the first one, instead make the mainline history urgly.
Each patch should work well individual without the following patch.

Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_server.c]: @xiaoxiang781216 Ok, I'll look into how to do what you ask!

File: examples/tcpblaster/tcpblaster_client.c (hunk 6 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_client.c]: @xiaoxiang781216 ok

Comment (xiaoxiang781216) [reply examples/tcpblaster/tcpblaster_client.c]: It's very easy, all is automatically with git rebase --interactive:
https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d
Once your finish to refine the patchset, you can run git push with -f flag to update the PR.

File: examples/tcpblaster/tcpblaster_server.c (hunk 5 above)
Comment (xiaoxiang781216) [reply examples/tcpblaster/tcpblaster_server.c]: It's very easy, all is automatically with git rebase --interactive:
https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d
Once your finish to refine the patchset, you can run git push with -f flag to update the PR.

File: examples/tcpblaster/tcpblaster_client.c (hunk 6 above)
Comment (adamfeuer) [reply examples/tcpblaster/tcpblaster_client.c]: @xiaoxiang781216 rebase against master and push -f, got it. Thanks. I will fix this tomorrow!

**Comment by adamfeuer** [comment]
@xiaoxiang781216 I made the requested improvements, rebased, and force-pushed to this branch. Will you let me know if there are more improvements you'd like to see?

**Comment by xiaoxiang781216** [comment]
@adamfeuer In the furture PR, it's better to remove the internal merge from the commit message.

**Comment by adamfeuer** [comment]
@xiaoxiang781216 Got it, will remove in future PRs.

**Pull request merged**